src/main/scala/io/iohk/ethereum/domain/SignedTransaction.scala

```diff
@@ -23,98 +23,93 @@ object SignedTransaction {
       Some(negativePointSign)
     } else if (pointSign == positivePointSign || pointSign == (Blockchain.chainId * 2 + newPositivePointSign).toByte) {
       Some(positivePointSign)
     } else {
       None
     }
-
-  lazy val recoveredPublicKey: Option[Array[Byte]] = recoveredPointSign.flatMap { p =>
-    ECDSASignature.recoverPubBytes(
-      new BigInteger(1, signatureRandom.toArray[Byte]),
-      new BigInteger(1, signature.toArray[Byte]),
-      ECDSASignature.recIdFromSignatureV(p),
-      bytesToSign
-    )
   }
 
-  lazy val syntacticValidity: Boolean = {
+  def getSender(tx: Transaction, signature: ECDSASignature, recoveredPointSign: Int): Option[Address] = {
+    val ECDSASignature(r, s, v) = signature
+    val bytesToSign: Array[Byte] = if (v == negativePointSign || v == positivePointSign) {
+      //global transaction
+      crypto.kec256(
+        rlpEncode(RLPList(
+          tx.nonce,
+          tx.gasPrice,
+          tx.gasLimit,
+          tx.receivingAddress.toArray,
+          tx.value,
+          tx.payload)))
+    } else {
+      //chain specific transaction
+      crypto.kec256(
+        rlpEncode(RLPList(
+          tx.nonce,
+          tx.gasPrice,
+          tx.gasLimit,
+          tx.receivingAddress.toArray,
+          tx.value,
+          tx.payload,
+          Config.Blockchain.chainId,
+          valueForEmptyR,
+          valueForEmptyS)))
+    }
 
-    import tx._
-    import Transaction._
+    val recoveredPublicKey: Option[Array[Byte]] =
+      ECDSASignature.recoverPubBytes(
+        new BigInteger(1, r.toByteArray),
+        new BigInteger(1, s.toByteArray),
+        ECDSASignature.recIdFromSignatureV(recoveredPointSign),
+        bytesToSign
+      )
 
-    def byteLength(b: BigInt): Int = b.toByteArray.length
+    for {
+      key <- recoveredPublicKey
+      addrBytes = crypto.kec256(key).slice(FirstByteOfAddress, LastByteOfAddress)
+      if addrBytes.length == Address.Length
+    } yield Address(addrBytes)
+  }
 
-    byteLength(nonce) <= NonceLength &&
-    //FIXME: the below doesn't make sense, an exception will be thrown if the condition is not met
-    //(receivingAddress.bytes.isEmpty || receivingAddress.bytes.length == AddressLength) &&
-    byteLength(gasLimit) <= GasLength &&
-    byteLength(gasPrice) <= GasLength &&
-    byteLength(value) <= ValueLength &&
-    signatureRandom.length <= ECDSASignature.RLength &&
-    signature.length <= ECDSASignature.SLength &&
-    recoveredSenderAddress.isDefined
+  def apply(tx: Transaction, pointSign: Byte, signatureRandom: ByteString, signature: ByteString): Option[SignedTransaction] = {
+    val txSignature = ECDSASignature(r = new BigInteger(1, signatureRandom.toArray), s = new BigInteger(1, signature.toArray), v = pointSign)
+    SignedTransaction(tx, txSignature)
   }
 
+  def apply(tx: Transaction, pointSign: Int, signatureRandom: ByteString, signature: ByteString): Option[SignedTransaction] = {
+    if(pointSign.toByte != pointSign)
```

Comment:
Good catch, I would just skip this condition. If pointSign is invalid, you will not be able to recover public key, and this will result in `None` anyway.